packages/react-dom/src/__tests__/ReactUpdates-test.js

```diff
@@ -1290,6 +1290,67 @@ describe('ReactUpdates', () => {
     expect(ops).toEqual(['Foo', 'Bar', 'Baz']);
   });
 
+  it('delays sync updates inside hidden subtrees in Concurrent Mode', () => {
+    const container = document.createElement('div');
+
+    function Baz() {
+      Scheduler.yieldValue('Baz');
+      return <p>baz</p>;
+    }
+
+    let setCounter;
+    function Bar() {
+      const [counter, _setCounter] = React.useState(0);
+      setCounter = _setCounter;
+      Scheduler.yieldValue('Bar');
+      return <p>bar {counter}</p>;
+    }
+
+    function Foo() {
+      Scheduler.yieldValue('Foo');
+      React.useEffect(() => {
+        Scheduler.yieldValue('Foo#effect');
+      });
+      return (
+        <div>
+          <div hidden={true}>
+            <Bar />
+          </div>
+          <Baz />
+        </div>
+      );
+    }
+
+    const root = ReactDOM.unstable_createRoot(container);
+    root.render(<Foo />);
+    expect(Scheduler).toFlushAndYieldThrough(
+      ['Foo', __DEV__ && 'Foo', 'Baz', 'Foo#effect'].filter(Boolean),
```

Comment:
😐

OK

🙂

Reply:
Better than making the test internal only just to flip this flag.

Reply:
(We should figure out something even better though. A bunch of tests don't run on bundles jut because of this inconvenience.)

Reply:
I would have done

```js
if (__DEV__) {
  expect(Scheduler).toFlushAndYieldThrough(['Foo', 'Foo', 'Baz', 'Foo#effect']);
} else {
  expect(Scheduler).toFlushAndYieldThrough(['Foo', 'Baz', 'Foo#effect']);
}
```